chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/impl/PendingSurveillanceManagerImpl.java

```diff
@@ -0,0 +1,728 @@
+            throws ValidationException, EntityCreationException, EntityRetrievalException {
+        if (file.isEmpty()) {
+            throw new ValidationException("You cannot upload an empty file!");
+        }
+
+        if (!file.getContentType().equalsIgnoreCase("text/csv")
+                && !file.getContentType().equalsIgnoreCase("application/vnd.ms-excel")) {
+            throw new ValidationException("File must be a CSV document.");
+        }
+
+        Integer surveillanceThresholdToProcessAsJob = getSurveillanceRecordThreshold();
+
+        //first we need to count how many surveillance records are in the file
+        //to know if we handle it normally or as a background job
+        String data = fileUtils.readFileAsString(file);
+
+        //This is a container used for 2 different result types...
+        SurveillanceUploadResult uploadResult = new SurveillanceUploadResult();
+
+        int numSurveillance = survUploadHelper.countSurveillanceRecords(data);
+        if (numSurveillance < surveillanceThresholdToProcessAsJob) {
+            uploadResult = processAsFile(file);
+        } else { //process as job
+            uploadResult = processUploadAsJob(data);
+        }
+        return uploadResult;
+    }
+
+    @Override
+    @Transactional
+    @PreAuthorize("@permissions.hasAccess(T(gov.healthit.chpl.permissions.Permissions).PENDING_SURVEILLANCE, "
+            + "T(gov.healthit.chpl.permissions.domains.PendingSurveillanceDomainPermissions).REJECT, "
+            + "#pendingSurveillanceId)")
+    public void rejectPendingSurveillance(final Long pendingSurveillanceId)
+            throws ObjectMissingValidationException, JsonProcessingException, EntityRetrievalException, EntityCreationException {
+
+        deletePendingSurveillance(pendingSurveillanceId, false);
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    @PreAuthorize("@permissions.hasAccess(T(gov.healthit.chpl.permissions.Permissions).PENDING_SURVEILLANCE, "
+            + "T(gov.healthit.chpl.permissions.domains.PendingSurveillanceDomainPermissions).GET_ALL)")
+    //@PostFilter("@permissions.hasAccess(T(gov.healthit.chpl.permissions.Permissions).PENDING_SURVEILLANCE, "
+    //        + "T(gov.healthit.chpl.permissions.domains.PendingSurveillanceDomainPermissions).GET_ALL, filterObject)")
+    public List<Surveillance> getAllPendingSurveillances() {
+        List<PendingSurveillanceEntity> pendingResults = survDao.getAllPendingSurveillance();
+        List<Surveillance> results = new ArrayList<Surveillance>();
+        if (pendingResults != null) {
+            for (PendingSurveillanceEntity pr : pendingResults) {
+                Surveillance surv = convertToDomain(pr);
+                results.add(surv);
+            }
+        }
+
+        //TODO: There is a problem using @PostFilter - some sort of ACL issue
+        //Until that is figured out...
+        Iterator<Surveillance> iterator = results.iterator();
```

Comment:
I kind of think we should figure it out if we're saying this is the new security framework. I feel like if we put it off this code will just end up staying here. I'd be okay with fixing it like in the next sprint if it's too much/frustrating for now, just don't want it to get forgotten about.

Reply:
I'll look at it for a couple of hours - if it takes much longer, we'll push it off.  I'll update where I am in this chat.

Reply:
Addressed